packages/oui-dropdown/src/dropdown.controller.js

```diff
@@ -0,0 +1,122 @@
+import Popper from 'popper.js'
+import { addBooleanParameter } from '@oui-angular/common/component-utils'
+
+const KEY_ESCAPE = 27
+
+export default class {
+  constructor ($attrs, $document, $element, $log, $scope) {
+    'ngInject'
+
+    this.$attrs = $attrs
+    this.$document = $document
+    this.$element = $element
+    this.$log = $log
+    this.$scope = $scope
+  }
+
+  $onInit () {
+    this.isDropdownOpen = false
+    this.hasFocus = false
+    this.currentFocusedElement = null
+    addBooleanParameter(this, 'arrow')
+    addBooleanParameter(this, 'start')
+    addBooleanParameter(this, 'end')
+
+    if (this.start && this.end) {
+      this.$log.warn('A dropdown cannot be positionned at the start and at the end at the same time.')
+    }
+
+    this.documentClickHandler = evt => {
+      if (evt &&
+        evt.type === 'click' &&
+        this.referenceElement.contains(evt.target)) {
+        return
+      }
+
+      this.referenceElement.focus()
+      this.$scope.$apply(() => this.closeDropdown())
```

Comment:
$scope.$apply, why ? :(

http://www.technofattie.com/2014/03/21/five-guidelines-for-avoiding-scope-soup-in-angular.html

Reply:
Because the event is managed natively. So the order comes from outside Angular. You cannot avoid using an $apply here.